### ISSUE TITLE
[camera] Preview has wrong orientation when using corner orientation button 

### ISSUE THREAD

**Issue by davidmartos96** [flutter/flutter#83603]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows
3. With a device that has portrait as its natural orientation start the app in portrait.
4. Rotate the device 90º counter clock wise and let the OS use the landscape orientation

**Expected results:** 
The preview should have the correct orientation

**Actual results:** 
The preview has the wrong orientation

Device used: Moto G5 Plus with Android 11
Camera plugin version: 0.8.1+1

https://user-images.githubusercontent.com/22084723/120077735-286be680-c0ac-11eb-9bbe-88a29166367c.mp4

<details>
  <summary>Logs</summary>

```
➜ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel dev, 2.3.0-12.1.pre, on openSUSE Tumbleweed
    5.12.0-2-default, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 4.2)
[✓] VS Code (version 1.56.2)
[✓] Connected device (2 available)

• No issues found!

```

</details>

```dart
import 'dart:async';
import 'package:flutter/material.dart';
import 'package:camera/camera.dart';

late List<CameraDescription> cameras;

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();

  cameras = await availableCameras();
  runApp(CameraApp());
}

class CameraApp extends StatefulWidget {
  @override
  _CameraAppState createState() => _CameraAppState();
}

class _CameraAppState extends State<CameraApp> {
  late CameraController controller;

  @override
  void initState() {
    super.initState();
    controller = CameraController(cameras[0], ResolutionPreset.max);
    controller.initialize().then((_) {
      if (!mounted) {
        return;
      }
      setState(() {});
    });
  }

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    if (!controller.value.isInitialized) {
      return Container();
    }
    return MaterialApp(
      home: Stack(
        children: [
          Positioned.fill(child: CameraPreview(controller)),
          Align(
            alignment: Alignment.bottomCenter,
            child: FloatingActionButton(
              onPressed: (){},
              child: Icon(Icons.camera),
            ),
          )
        ],
      ),
    );
  }
}

```

**Comment by TahaTesser** [comment]
Hi @davidmartos96 
I see there's an open issue addressing the case you described.
Please follow up on that issue, Closing as duplicate of https://github.com/flutter/flutter/issues/39669
If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by TahaTesser** [comment]
Hi @davidmartos96 
This seems to be working for me 

You need to unlock portrait mode 
Here is the quick demo with dash as the subject using [official sample](https://pub.dev/packages/video_player/example)


### Portrait locked

![ezgif com-gif-maker](https://user-images.githubusercontent.com/48603081/120209515-72350800-c237-11eb-9a62-65ffe4a115d6.gif)


### Portrait unlocked

![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/48603081/120210929-0489db80-c239-11eb-97c6-53644e722acf.gif)


Can you please provide device details
Thank you

**Comment by davidmartos96** [comment]
@TahaTesser Thanks for reopening! This is what I see when having it unlocked.


https://user-images.githubusercontent.com/22084723/120218036-76622500-c239-11eb-982b-9752fa570f30.mp4

I even followed the codelab for the CameraX to see if there could be something wrong with my device but the camera rotation works as expected with it. https://developer.android.com/codelabs/camerax-getting-started#0

**Comment by TahaTesser** [comment]
Hi @davidmartos96 

> Device used: Moto G5 Plus with Android 11
> Camera plugin version: 0.8.1+1

The issue could in the `camera` package but reproduces only this device, `camerax` comparison won't matter.  A better way to be sure is to try different with `camera` and see if you can reproduce the issue.  You can see in my response, it works fine

**Comment by davidmartos96** [comment]
@TahaTesser Actually, I'm seeing the same problem on a Samsung Galaxy Grand Prime (SM-G531F) Android 5.1.1.
But it works as expected in another device I have (Xiaomi Mi A1)

Why not compare it with the camerax library? A far as I know, in the end is an abstraction over `Camera2`, same as the Flutter `camera` package.

**Comment by davidmartos96** [comment]
The cameraX library has this class for dealing with device rotations, but it's not exported publicly. Maybe something similar could be accomplished 
https://cs.android.com/androidx/platform/frameworks/support/+/androidx-master-release:camera/camera-view/src/main/java/androidx/camera/view/PreviewTransformation.java

**Comment by TahaTesser** [comment]
Hi @davidmartos96 
I just tried a Samsung device (M02s running Android 11), orientation is working as expected. So far I've tried Oneplus 8, Xioami K20 pro, Samsung M02s, none of these devices has this issue if the orientation is unlocked.  Preview in https://github.com/flutter/flutter/issues/83603#issuecomment-851542506

Since you can experience it, I'm gonna label this
Thank you

**Comment by davidmartos96** [comment]
I've just tested the basic Camera2 demo from the official Android samples and it works correctly.
https://github.com/android/camera-samples/tree/main/Camera2Basic
Maybe there is a small edge case missing in the plugin implementation.

**Comment by davidmartos96** [comment]
@TahaTesser After some investigation I found out that the problem is not really my device. It has to do with the rotation lock buttons that come with Android 9.0 and up. If I unlock the rotation (free rotation) it works as expected on my device. But, if I lock the rotation, and I manually rotate pressing the Android UI buttons, it reproduces the error from above.
This happens as well on the emulator. Tested the Pixel C tablet and the Pixel 2 phone emulators. 

Could you try to reproduce it on your devices?

![image](https://user-images.githubusercontent.com/22084723/121212305-81423880-c87d-11eb-8cce-8df08428fd82.png)

**Comment by TahaTesser** [comment]
Hi @davidmartos96 
That's an excellent find, I just tested the issue. with the Android orientation button, I can confirm the finally reproduces

### Tested on Oneplus 8

| System orientation | Orientation button | 
| --------------- | --------------- |
<img src="https://user-images.githubusercontent.com/48603081/121688539-408c2e80-cacc-11eb-9bef-df01971d298e.gif" height="450" /> | <img src="https://user-images.githubusercontent.com/48603081/121687593-3584ce80-cacb-11eb-8d33-46dfe4fceb1d.gif" height="450" /> |













### Sample
https://pub.dev/packages/image_picker/example

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.1, on Microsoft Windows [Version 10.0.19043.1052], locale en-US)
    • Flutter version 2.2.1 at C:\Users\Taha\Code\flutter_stable
    • Framework revision 02c026b03c (2 weeks ago), 2021-05-27 12:24:44 -0700
    • Engine revision 0fdb562ac8
    • Dart version 2.13.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\openjdk\bin\java.exe
    • Java version OpenJDK Runtime Environment (build 1.8.0-262-b10)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.0)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31321.278
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.1
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7233

[✓] VS Code (version 1.57.0)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.23.0

[✓] Connected device (4 available)
    • IN2011 (mobile)   • c9d8ee0c • android-arm64  • Android 11 (API 30)
    • Windows (desktop) • windows  • windows-x64    • Microsoft Windows [Version 10.0.19043.1052]
    • Chrome (web)      • chrome   • web-javascript • Google Chrome 91.0.4472.101
    • Edge (web)        • edge     • web-javascript • Microsoft Edge 91.0.864.41

! Doctor found issues in 1 category.
```

</details>

**Comment by Adamaq01** [comment]
Hi, any news on a solution for that issue ?

**Comment by BeMacized** [comment]
I no longer seem to be able to reproduce this on flutter/plugins#4059, so I am guessing this issue will be fixed once the android refactor has been fully merged.

**Comment by BeMacized** [comment]
Closing this as this fix has been merged with flutter/plugins#4059.

**Comment by davidmartos96** [comment]
Everything works with the latest update. Thank you for the work on the PR @BeMacized !
